app/src/componentsV2/BlockScreen/hooks/useIsUserBlocked.ts

```diff
@@ -0,0 +1,109 @@
+import firebaseApp from "firebase";
+import { getDatabase, onValue, ref } from "firebase/database";
+import { doc, getFirestore, onSnapshot } from "firebase/firestore";
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/slices/global/user/selectors";
+
+export enum BlockType {
+  GRR = "grr",
+  COMPLIANCE_ISSUE = "compliance-issue",
+}
+
+export type BlockConfig = {
+  [key in BlockType]?: {
+    isBlocked: boolean;
+    reason?: string;
+    metadata?: Record<string, any>;
+  };
+};
+
+export const useIsUserBlocked = () => {
+  const user = useSelector(getUserAuthDetails);
+  const isLoggedIn = user?.loggedIn;
+  const uid = user?.details?.profile?.uid;
+  const email = user?.details?.profile?.email;
+
+  const [domainBlockConfig, setDomainBlockConfig] = useState<BlockConfig>({});
+  const [userBlockConfig, setUserBlockConfig] = useState<BlockConfig>({});
+  const [finalBlockConfig, setFinalBlockConfig] = useState<BlockConfig>({});
+
+  useEffect(() => {
+    if (!isLoggedIn) {
+      return;
+    }
+
+    const db = getFirestore(firebaseApp);
+    const unsubscribeListener = onSnapshot(doc(db, "users", uid), (doc) => {
+      if (doc.exists()) {
+        const userDetails = doc.data();
+        setUserBlockConfig(userDetails?.["block-config"] || {});
+      }
+    });
+
+    return () => {
+      unsubscribeListener?.();
+    };
+  }, [isLoggedIn, uid]);
+
+  useEffect(() => {
+    if (!isLoggedIn) {
+      return;
+    }
+
+    const emailDomain = email?.split("@")[1];
+    const emailDomainKey = emailDomain?.replaceAll(".", "_dot_");
+    const rdb = getDatabase();
+    const domainBlockConfigRef = ref(rdb, `globalBlockConfig/domain/${emailDomainKey}`);
```

Comment:
Are we going to handle this domains manually? what if we get a set of domains to be block, do we have any script which write here?

Reply:
Manually for now.